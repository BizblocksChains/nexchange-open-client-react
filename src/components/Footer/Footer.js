import React from 'react';
import { I18n, Trans } from 'react-i18next';
import { Link } from 'react-router-dom';
import styles from './Footer.scss';

const Footer = () => (
  <I18n ns="translations">
    {t => (
      <footer className={styles.container}>
        <div className="container">
          <div className="row">
            <div className="col-xs-12 col-sm-8 col-md-6">
              <Trans i18nKey="footer.3">
                <p className={styles.address}>
                  All rights reserved, YOA LTD 2016-2017,<br />England & Wales{' '}
                  <a href="https://beta.companieshouse.gov.uk/company/10009845" rel="noopener noreferrer" target="_blank">
                    registered company No. 10009845
                  </a>
                </p>
              </Trans>
            </div>

            <div className="col-xs-12 col-sm-4 col-md-6">
              <ul className={styles.links}>
                <li>
                  <Link to="/terms-and-conditions">{t('footer.1')}</Link>
                </li>
                <li>
                  <Link to="/privacy">{t('footer.2')}</Link>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </footer>
    )}
  </I18n>
);

export default Footer;
