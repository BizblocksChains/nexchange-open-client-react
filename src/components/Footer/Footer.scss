.container {
  padding: 20px 0 15px 0;
  width: 100%;
  min-height: 114px;
  background: #fff;

  @media (min-width: $screen-sm-min) {
    position: absolute;
    bottom: 0;
    height: 114px;
  }

  p {
    font-size: 14px;
  }

  .address {
    @media (min-width: $screen-sm-min) {
      margin-top: 15px;
    }
  }

  .links {
    padding: 0;
    margin: 0;

    @media (min-width: $screen-sm-min) {
      margin-top: 12px;
      float: right;
    }

    @media (min-width: $screen-md-min) {
      margin-top: 24px;
    }

    li {
      a {
        padding: 0;
        color: $gray;
        font-size: 14px;
        font-weight: 300;
      }

      @media (min-width: $screen-md-min) {
        padding-left: 30px;
      }
    }
  }
}
