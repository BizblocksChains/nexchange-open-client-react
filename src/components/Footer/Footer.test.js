import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

describe('Footer', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<Footer />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
