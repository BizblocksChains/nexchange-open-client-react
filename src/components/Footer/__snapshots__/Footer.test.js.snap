// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
>
        <div
                className="row"
        >
                <div
                        className="col-xs-12 col-sm-8 col-md-6"
                >
                        <Trans
                                i18nKey="footer.3"
                        >
                                <p
                                        className="address"
                                >
                                        All rights reserved, YOA LTD 2016-2017,
                                        <br />
                                        England & Wales
                                         
                                        <a
                                                href="https://beta.companieshouse.gov.uk/company/10009845"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                        >
                                                registered company No. 10009845
                                        </a>
                                </p>
                        </Trans>
                </div>
                <div
                        className="col-xs-12 col-sm-4 col-md-6"
                >
                        <ul
                                className="links"
                        >
                                <li>
                                        <Link
                                                replace={false}
                                                to="/terms-and-conditions"
                                        >
                                                footer.1
                                        </Link>
                                </li>
                                <li>
                                        <Link
                                                replace={false}
                                                to="/privacy"
                                        >
                                                footer.2
                                        </Link>
                                </li>
                        </ul>
                </div>
        </div>
</div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <div
                    className="col-xs-12 col-sm-8 col-md-6"
          >
                    <Trans
                              i18nKey="footer.3"
                    >
                              <p
                                        className="address"
                              >
                                        All rights reserved, YOA LTD 2016-2017,
                                        <br />
                                        England & Wales
                                         
                                        <a
                                                  href="https://beta.companieshouse.gov.uk/company/10009845"
                                                  rel="noopener noreferrer"
                                                  target="_blank"
                                        >
                                                  registered company No. 10009845
                                        </a>
                              </p>
                    </Trans>
          </div>
          <div
                    className="col-xs-12 col-sm-4 col-md-6"
          >
                    <ul
                              className="links"
                    >
                              <li>
                                        <Link
                                                  replace={false}
                                                  to="/terms-and-conditions"
                                        >
                                                  footer.1
                                        </Link>
                              </li>
                              <li>
                                        <Link
                                                  replace={false}
                                                  to="/privacy"
                                        >
                                                  footer.2
                                        </Link>
                              </li>
                    </ul>
          </div>
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-xs-12 col-sm-8 col-md-6"
>
              <Trans
                            i18nKey="footer.3"
              >
                            <p
                                          className="address"
                            >
                                          All rights reserved, YOA LTD 2016-2017,
                                          <br />
                                          England & Wales
                                           
                                          <a
                                                        href="https://beta.companieshouse.gov.uk/company/10009845"
                                                        rel="noopener noreferrer"
                                                        target="_blank"
                                          >
                                                        registered company No. 10009845
                                          </a>
                            </p>
              </Trans>
</div>,
            <div
              className="col-xs-12 col-sm-4 col-md-6"
>
              <ul
                            className="links"
              >
                            <li>
                                          <Link
                                                        replace={false}
                                                        to="/terms-and-conditions"
                                          >
                                                        footer.1
                                          </Link>
                            </li>
                            <li>
                                          <Link
                                                        replace={false}
                                                        to="/privacy"
                                          >
                                                        footer.2
                                          </Link>
                            </li>
              </ul>
</div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Trans
                i18nKey="footer.3"
>
                <p
                                className="address"
                >
                                All rights reserved, YOA LTD 2016-2017,
                                <br />
                                England & Wales
                                 
                                <a
                                                href="https://beta.companieshouse.gov.uk/company/10009845"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                >
                                                registered company No. 10009845
                                </a>
                </p>
</Trans>,
              "className": "col-xs-12 col-sm-8 col-md-6",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <p
                  className="address"
>
                  All rights reserved, YOA LTD 2016-2017,
                  <br />
                  England & Wales
                   
                  <a
                                    href="https://beta.companieshouse.gov.uk/company/10009845"
                                    rel="noopener noreferrer"
                                    target="_blank"
                  >
                                    registered company No. 10009845
                  </a>
</p>,
                "i18nKey": "footer.3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "All rights reserved, YOA LTD 2016-2017,",
                    <br />,
                    "England & Wales",
                    " ",
                    <a
                      href="https://beta.companieshouse.gov.uk/company/10009845"
                      rel="noopener noreferrer"
                      target="_blank"
>
                      registered company No. 10009845
</a>,
                  ],
                  "className": "address",
                },
                "ref": null,
                "rendered": Array [
                  "All rights reserved, YOA LTD 2016-2017,",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "England & Wales",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "registered company No. 10009845",
                      "href": "https://beta.companieshouse.gov.uk/company/10009845",
                      "rel": "noopener noreferrer",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": "registered company No. 10009845",
                    "type": "a",
                  },
                ],
                "type": "p",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="links"
>
                <li>
                                <Link
                                                replace={false}
                                                to="/terms-and-conditions"
                                >
                                                footer.1
                                </Link>
                </li>
                <li>
                                <Link
                                                replace={false}
                                                to="/privacy"
                                >
                                                footer.2
                                </Link>
                </li>
</ul>,
              "className": "col-xs-12 col-sm-4 col-md-6",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <Link
                                        replace={false}
                                        to="/terms-and-conditions"
                    >
                                        footer.1
                    </Link>
</li>,
                  <li>
                    <Link
                                        replace={false}
                                        to="/privacy"
                    >
                                        footer.2
                    </Link>
</li>,
                ],
                "className": "links",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/terms-and-conditions"
>
                      footer.1
</Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "footer.1",
                      "replace": false,
                      "to": "/terms-and-conditions",
                    },
                    "ref": null,
                    "rendered": "footer.1",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/privacy"
>
                      footer.2
</Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "footer.2",
                      "replace": false,
                      "to": "/privacy",
                    },
                    "ref": null,
                    "rendered": "footer.2",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          <div
                    className="row"
          >
                    <div
                              className="col-xs-12 col-sm-8 col-md-6"
                    >
                              <Trans
                                        i18nKey="footer.3"
                              >
                                        <p
                                                  className="address"
                                        >
                                                  All rights reserved, YOA LTD 2016-2017,
                                                  <br />
                                                  England & Wales
                                                   
                                                  <a
                                                            href="https://beta.companieshouse.gov.uk/company/10009845"
                                                            rel="noopener noreferrer"
                                                            target="_blank"
                                                  >
                                                            registered company No. 10009845
                                                  </a>
                                        </p>
                              </Trans>
                    </div>
                    <div
                              className="col-xs-12 col-sm-4 col-md-6"
                    >
                              <ul
                                        className="links"
                              >
                                        <li>
                                                  <Link
                                                            replace={false}
                                                            to="/terms-and-conditions"
                                                  >
                                                            footer.1
                                                  </Link>
                                        </li>
                                        <li>
                                                  <Link
                                                            replace={false}
                                                            to="/privacy"
                                                  >
                                                            footer.2
                                                  </Link>
                                        </li>
                              </ul>
                    </div>
          </div>
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <div
                        className="col-xs-12 col-sm-8 col-md-6"
            >
                        <Trans
                                    i18nKey="footer.3"
                        >
                                    <p
                                                className="address"
                                    >
                                                All rights reserved, YOA LTD 2016-2017,
                                                <br />
                                                England & Wales
                                                 
                                                <a
                                                            href="https://beta.companieshouse.gov.uk/company/10009845"
                                                            rel="noopener noreferrer"
                                                            target="_blank"
                                                >
                                                            registered company No. 10009845
                                                </a>
                                    </p>
                        </Trans>
            </div>
            <div
                        className="col-xs-12 col-sm-4 col-md-6"
            >
                        <ul
                                    className="links"
                        >
                                    <li>
                                                <Link
                                                            replace={false}
                                                            to="/terms-and-conditions"
                                                >
                                                            footer.1
                                                </Link>
                                    </li>
                                    <li>
                                                <Link
                                                            replace={false}
                                                            to="/privacy"
                                                >
                                                            footer.2
                                                </Link>
                                    </li>
                        </ul>
            </div>
</div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-xs-12 col-sm-8 col-md-6"
>
                <Trans
                                i18nKey="footer.3"
                >
                                <p
                                                className="address"
                                >
                                                All rights reserved, YOA LTD 2016-2017,
                                                <br />
                                                England & Wales
                                                 
                                                <a
                                                                href="https://beta.companieshouse.gov.uk/company/10009845"
                                                                rel="noopener noreferrer"
                                                                target="_blank"
                                                >
                                                                registered company No. 10009845
                                                </a>
                                </p>
                </Trans>
</div>,
              <div
                className="col-xs-12 col-sm-4 col-md-6"
>
                <ul
                                className="links"
                >
                                <li>
                                                <Link
                                                                replace={false}
                                                                to="/terms-and-conditions"
                                                >
                                                                footer.1
                                                </Link>
                                </li>
                                <li>
                                                <Link
                                                                replace={false}
                                                                to="/privacy"
                                                >
                                                                footer.2
                                                </Link>
                                </li>
                </ul>
</div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Trans
                  i18nKey="footer.3"
>
                  <p
                                    className="address"
                  >
                                    All rights reserved, YOA LTD 2016-2017,
                                    <br />
                                    England & Wales
                                     
                                    <a
                                                      href="https://beta.companieshouse.gov.uk/company/10009845"
                                                      rel="noopener noreferrer"
                                                      target="_blank"
                                    >
                                                      registered company No. 10009845
                                    </a>
                  </p>
</Trans>,
                "className": "col-xs-12 col-sm-8 col-md-6",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <p
                    className="address"
>
                    All rights reserved, YOA LTD 2016-2017,
                    <br />
                    England & Wales
                     
                    <a
                                        href="https://beta.companieshouse.gov.uk/company/10009845"
                                        rel="noopener noreferrer"
                                        target="_blank"
                    >
                                        registered company No. 10009845
                    </a>
</p>,
                  "i18nKey": "footer.3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "All rights reserved, YOA LTD 2016-2017,",
                      <br />,
                      "England & Wales",
                      " ",
                      <a
                        href="https://beta.companieshouse.gov.uk/company/10009845"
                        rel="noopener noreferrer"
                        target="_blank"
>
                        registered company No. 10009845
</a>,
                    ],
                    "className": "address",
                  },
                  "ref": null,
                  "rendered": Array [
                    "All rights reserved, YOA LTD 2016-2017,",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    "England & Wales",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "registered company No. 10009845",
                        "href": "https://beta.companieshouse.gov.uk/company/10009845",
                        "rel": "noopener noreferrer",
                        "target": "_blank",
                      },
                      "ref": null,
                      "rendered": "registered company No. 10009845",
                      "type": "a",
                    },
                  ],
                  "type": "p",
                },
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ul
                  className="links"
>
                  <li>
                                    <Link
                                                      replace={false}
                                                      to="/terms-and-conditions"
                                    >
                                                      footer.1
                                    </Link>
                  </li>
                  <li>
                                    <Link
                                                      replace={false}
                                                      to="/privacy"
                                    >
                                                      footer.2
                                    </Link>
                  </li>
</ul>,
                "className": "col-xs-12 col-sm-4 col-md-6",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li>
                      <Link
                                            replace={false}
                                            to="/terms-and-conditions"
                      >
                                            footer.1
                      </Link>
</li>,
                    <li>
                      <Link
                                            replace={false}
                                            to="/privacy"
                      >
                                            footer.2
                      </Link>
</li>,
                  ],
                  "className": "links",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/terms-and-conditions"
>
                        footer.1
</Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "footer.1",
                        "replace": false,
                        "to": "/terms-and-conditions",
                      },
                      "ref": null,
                      "rendered": "footer.1",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/privacy"
>
                        footer.2
</Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "footer.2",
                        "replace": false,
                        "to": "/privacy",
                      },
                      "ref": null,
                      "rendered": "footer.2",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
